tools/mock_node/src/net.rs

```diff
@@ -0,0 +1,278 @@
+use async_trait::async_trait;
+use bytes::buf::{Buf, BufMut};
+use bytes::BytesMut;
+use near_crypto::SecretKey;
+use near_network::types::{Encoding, PeerMessage};
+use near_network_primitives::types::{
+    AccountOrPeerIdOrHash, RawRoutedMessage, RoutedMessageBody, TcpListener, TcpStream, TcpSys,
+};
+use near_primitives::network::PeerId;
+use std::io;
+use std::net::SocketAddr;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex, RwLock, Weak};
+use std::task::{Context, Poll, Waker};
+use std::time::Duration;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+use crate::{MockNet, MockPeer};
+
+struct ReadBuffer {
+    buf: BytesMut,
+    reader: Option<Waker>,
+}
+
+struct MockTcpStreamInner {
+    local_addr: SocketAddr,
+    peer_addr: SocketAddr,
+    response_delay: Duration,
+    read_buf: Arc<RwLock<ReadBuffer>>,
```

Comment:
yeah true

Reply:
in fact theres not gonna be any contention anyway since it's all happening in the same thread (we spawn with `actix::spawn`, which does `tokio::task::spawn_local`). sadly cant replace it w faster single threaded stuff (like `Rc<RefCell<>>` bc it needs to be `Send` since it gets passed around in actix messages